### PR TITLE
added config for connecting to postgres, auth, swagger

### PR DESCRIPTION
**Pull request by carlov20**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -4,11 +4,34 @@ plugins {
...
+
+  implementation("io.opentelemetry:opentelemetry-api:1.22.0")
+
+  runtimeOnly("org.flywaydb:flyway-core")
+  runtimeOnly("org.postgresql:r2dbc-postgresql:0.9.3.RELEASE")
+  runtimeOnly("org.postgresql:postgresql:42.5.1")
+
+  testImplementation("org.jetbrains.kotlinx:kotlinx-coroutines-test:1.6.4")
+
+  testImplementation("com.github.tomakehurst:wiremock-standalone:2.27.2")
```
Comment (sp-watson) [inline build.gradle.kts:34]: Are we needing wiremock - I was thinking mock server would be OK?!

File: helm_deploy/hmpps-staff-lookup-service/values.yaml [hunk 2]
```diff
@@ -43,3 +48,4 @@ generic-service:
 
 generic-prometheus-alerts:
   targetApplication: hmpps-staff-lookup-service
+  alertSeverity: hmpps-tier
```
Comment (sp-watson) [inline helm_deploy/hmpps-staff-lookup-service/values.yaml:51]: Hmm.. we probably need another slack channel? I don't know..

File: renovate.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (sp-watson) [inline renovate.json:1]: Yep, good

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstafflookupservice/config/ClientTrackingWebFilter.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import reactor.core.publisher.Mono
+import java.text.ParseException
+
+@Component
+class ClientTrackingWebFilter : WebFilter {
+  private val bearer = "Bearer "
+  private val clientIdKey = AttributeKey.stringKey("clientId")
+  private val usernameKey = AttributeKey.stringKey("username")
+
+  override fun filter(exchange: ServerWebExchange, chain: WebFilterChain): Mono<Void> {
```
Comment (sp-watson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstafflookupservice/config/ClientTrackingWebFilter.kt:22]: Hmm.. that's a bit quirky when we are using coroutines.. ah well

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstafflookupservice/config/HmppsStaffLookupServiceExceptionHandler.kt [hunk 5]
```diff
@@ -7,36 +7,41 @@ import org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR
...
 import org.springframework.web.bind.annotation.ExceptionHandler
 import org.springframework.web.bind.annotation.RestControllerAdvice
+import reactor.core.publisher.Mono
 import javax.validation.ValidationException
 
 @RestControllerAdvice
 class HmppsStaffLookupServiceExceptionHandler {
   @ExceptionHandler(ValidationException::class)
-  fun handleValidationException(e: Exception): ResponseEntity<ErrorResponse> {
+  fun handleValidationException(e: Exception): Mono<ResponseEntity<ErrorResponse>> {
```
Comment (sp-watson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstafflookupservice/config/HmppsStaffLookupServiceExceptionHandler.kt:16]: I wouldn't have thought we needed the Mono if we are using coroutines?

File: src/main/resources/application.yml [hunk 6]
```diff
@@ -49,3 +68,8 @@ management:
     info:
       cache:
         time-to-live: 2000ms
+
+azure:
+  application-insights:
+    web:
+      enabled: false
```
Comment (sp-watson) [inline src/main/resources/application.yml:75]: Not sure what this means?!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsstafflookupservice/wiremock/HmppsAuthMockServer.kt [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock
+import com.github.tomakehurst.wiremock.http.HttpHeader
+import com.github.tomakehurst.wiremock.http.HttpHeaders
+import org.junit.jupiter.api.extension.AfterAllCallback
+import org.junit.jupiter.api.extension.BeforeAllCallback
+import org.junit.jupiter.api.extension.BeforeEachCallback
+import org.junit.jupiter.api.extension.ExtensionContext
+
+class HmppsAuthApiExtension : BeforeAllCallback, AfterAllCallback, BeforeEachCallback {
```
Comment (sp-watson) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsstafflookupservice/wiremock/HmppsAuthMockServer.kt:12]: I am a bit confused by all this auth stuff - we don't need to act as an HMPPS auth client. But maybe this is to accept incoming calls?

**Review by sp-watson — COMMENTED** [review]
I need some clarification on some of these, cos there is quite a lot in here

File: build.gradle.kts [hunk 8]
```diff
@@ -4,11 +4,34 @@ plugins {
 }
 
 configurations {
+  implementation { exclude(module = "spring-boot-starter-web") }
```
Comment (sp-watson) [inline build.gradle.kts:7]: Oh .. are these here because of clashes?

File: build.gradle.kts (hunk 1 above)
Comment (carlov20) [reply build.gradle.kts:34]: I think I've added too much here, don't think we need this for now, was used with https://github.com/ministryofjustice/hmpps-staff-lookup-service/blob/d06d66478b3bfed6f9feb7afaf1cd6f6e594c8be/src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsstafflookupservice/wiremock/HmppsAuthMockServer.kt but think that will be for clients

File: helm_deploy/hmpps-staff-lookup-service/values.yaml (hunk 2 above)
Comment (carlov20) [reply helm_deploy/hmpps-staff-lookup-service/values.yaml:51]: this is not named the greatest, this means go to slack channel manage_a_workforce_dev

Comment (sp-watson) [reply helm_deploy/hmpps-staff-lookup-service/values.yaml:51]: It's just a mapping to our channel

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsstafflookupservice/wiremock/HmppsAuthMockServer.kt (hunk 7 above)
Comment (sp-watson) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsstafflookupservice/wiremock/HmppsAuthMockServer.kt]: We need it anyway. These extensions are like cross-cutting concerns, which apply set actions to all tests without messing up the test code.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstafflookupservice/config/ClientTrackingWebFilter.kt (hunk 4 above)
Comment (sp-watson) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstafflookupservice/config/ClientTrackingWebFilter.kt:22]: I suppose these return values are for Spring to process, so it's fine.

File: build.gradle.kts (hunk 1 above)
Comment (sp-watson) [reply build.gradle.kts]: Going to remove for now - we will use Mock Server

File: build.gradle.kts (hunk 8 above)
Comment (sp-watson) [reply build.gradle.kts:7]: Confirmed - specifically this is to prevent non-reactive code form been executed

**Review by sp-watson — APPROVED** [review]

**Pull request merged**